src/gc/env/gcenv.ee.h

```diff
@@ -78,7 +80,7 @@ class GCToEEInterface
 
     static void GcEnumAllocContexts(enum_alloc_context_func* fn, void* param);
 
-    static void AttachCurrentThread(); // does not acquire thread store lock
+    static bool CreateBackgroundThread(Thread** thread, GCBackgroundThreadFunction threadStart, void* arg);
```

Comment:
I know it is odd, but since GCToEEInterface doesn't know anything about gc_heap, it has no way to make sure the pointer is stored in the gc_heap::bgc_thread before the thread function starts unless we pass it the location to store it.
A way to solve that would be to change the bgc_thread_stub argument to a struct containing both the gc_heap pointer (as void* to make it transparent) and the Thread*. Then the stub can set the heap->bgc_thread as the first thing. 
I'll modify it that way.